### PR TITLE
Issue 4 user can add spells to their list

### PR DESCRIPTION
**Pull request by jnsangalang**

### CODE REVIEW

**Comment by jnsangalang** [comment]
issue 4 mobilefront end project
issue 4 desktop front end project

File: ts/main.ts [hunk 1]
```diff
@@ -401,8 +432,18 @@ function renderSpell(spellData: SpellInformation): HTMLDivElement {
   $spellTitle.classList.add('title-name');
   $spellTitle.textContent = spellData.name;
 
+  if (data.spellEdit) {
+    $divSpell.dataset.spellId = data.spellEdit.spellId.toString();
+  }
+  if (!data.spellEdit) {
+    $divSpell.dataset.spellId = data.nextSpellId.toString();
+  }
```
Comment (shawnkost) [inline ts/main.ts]: Why not just do an `else` statement rather than another `if` statement?

File: ts/main.ts [hunk 2]
```diff
@@ -596,5 +640,138 @@ function renderSpell(spellData: SpellInformation): HTMLDivElement {
...
+      }
+    }
+  }
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  for (let i = 0; i < data.actualSpellList.length; i++) {
+    if (data.actualSpellList[i] !== null) {
+      if (previousDataJSON !== null) {
+        data = JSON.parse(previousDataJSON);
```
Comment (shawnkost) [inline ts/main.ts]: This already runs inside of `data.ts`, so this shouldn't be needed

**Review by shawnkost — COMMENTED** [review]
Looks great Johan, just a couple comments here!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**